### PR TITLE
Train classifier

### PR DESCRIPTION
**Pull request by irodriguezclaveria**
Hi all, 

This pull request is related with the task #4 and task #5, I take advantage of the preprocessing data where we are using the CountVectorizer to create the prediction, take into account that here I've used the TfidfTransformer and MultinomialNB to create the predictions, this code is prototype and we should align and merge with the rest of the code.

Take into account the following pending points:

- That just is a test with a single example of tweet, is a prototype.
- In the next task #6 we should to associate this code with the tasks #7, #8....
- Is pending to include the documentation.
- The name of functions and files can be change when we align all code to check a correct performance
- I need to do more test in deep to check that all work correctly and check the performance, just I want to share the way that I take.
- If you know a better way, please let me know.
- I used the CrisisT6, for the training_data, I have to include the bibliography about that and we can include more files, that is just a test. 

Any comment, please, let me know.

### CODE REVIEW

**Comment by ulie50** [comment]
I think on the next spin or Cohort following enhancements cold be done:
1) Implement calculation of Accuracy, Precision, Recall and F1 score
2)  make train-test splitting in order to calculate Accuracy, Precision, Recall and F1 score
3) Develop other models (Logistic Regression, CNN or RNN) to compare and choose best suitable classification model for this case (with estimators written above).
I looked into results and some results seem questionable (see below), but we need to compare how other would perform with estimators.
_some results:_
Some short game (@ Kinsmen Pitch n Putt) http://t.co/2dV5scc5G9 --> Floods
The studio I record at flooded... #LNSDV1 might be pushed to a later date :( --> Floods
having coffee in Bowness while the rest of the world thinks we are 'under water' @ bowness http://t.co/xSiBPVz7T4 --> Floods
I shouldn't read anything without my glasses anymore.  Math = Meth and Magic Mouse = Magic Mike. --> Hurricane

**Comment by irodriguezclaveria** [comment]
Hi @ulie50 ,

Please check my comment in the pull request:

I need to do more test in deep to check that all work correctly and check the performance, just I want to share the way that I take.

This pull request is not finish.

Thanks.

**Comment by ulie50** [comment]
Hi @irodriguezclaveria ,

I read your comment and I know that is not finished :) You just did not write how you want to test it and that you are going to do splitting, so I wanted to share my thoughts how you could test and  what could be developed in parallel or on the next spin/cohort (moreover you wrote that you welcome sharing some ideas and )

**Comment by irodriguezclaveria** [comment]
> Hi @irodriguezclaveria ,
> 
> I read your comment and I know that is not finished :) You just did not write how you want to test it and that you are going to do splitting, so I wanted to share my thoughts how you could test and what could be developed in parallel or on the next spin/cohort (moreover you wrote that you welcome sharing some ideas and )

Perfect I will follow your advice and ideas :)

**Comment by irodriguezclaveria** [comment]
> 3\. Some short game (@ Kinsmen Pitch n Putt) http://t.co/2dV5scc5G9 --> Floods

Hi @ulie50,

I check the example that you say about the prediction model, I understand that when you extract the information from twitter, you include a filter of disasters, is that correct?

This example that you say, are real data extracted from twitter with disaster filter (I have to check you pull request), because if that is not real data, maybe is check with real cases not with corner cases that maybe never happens.

About this point:

make train-test splitting in order to calculate Accuracy, Precision, Recall and F1 score

Sorry, but I don't understand very well what do you want to do here "make train-test splitting " :), the training data is already splitting ... 

The other point:

Implement calculation of Accuracy, Precision, Recall and F1 score
 
I understand that this point is just to test it, is it correct?

**Comment by ulie50** [comment]
Hi Irene @irodriguezclaveria ,

Do you mean that you used the snippet from Front_End_Viz for Twitter data fetching to get the data? Currently it gets data there by hashtag: I just used this principium because it was used in other cohorts, but may be it is not the perfect one because some disaster tweets could be without hashtags.  Plus, some advertisement tweets get hashtags of disasters (may be they want they tweets to be seen so by many people, who knows). All in all the tweets extracted from there not always related to disaster.
So may be for classification we need to include one more category, something like "not disaster"/"other"/"not realted"?


I do not know if we talk about the same, I apologize if I could not find it in your code, so I will explain what I meant. When you create a model (doesnt matter which: regression, classification, NN and so on) you should split your data minimum on 2 parts: train and test data (there can be more splits, like additional validation set). With train set you train your model. But to get understanding how good your model is, you need to estimate it and for that purpose you should use test set, so you will see how your model performs on unseen data.

So it comes to your next question:  we need to estimate how good is the model we have with Accuracy, Precision, Recall and F1 score. I mean how else you can know how well it performed, to compare different tweets and so one - one hell of the job:) Moreover, it could be the case that another approach is better ( e.g. Logistic Regression) , we can now it by comparing estimators. 
NB: For those Estimators we have to label data => So we need to prepare datasets where we read and label if it is flood related or not.
I had somewhere a link with article related to that, I will share as soon as I find :)

**Comment by ulie50** [comment]
Okay, here is not written about train-test split, but entire process of tweet's classification is described: https://medium.com/@martinpella/customers-tweets-classification-41cdca4e2de 

But it is very good described for what to split data on different sets, for what to estimate models and how, what thoe estimators mean and many other data scince concepts in the book "Hands-On Machine Learning with Scikit-Learn, Keras, and TensorFlow",  ...and it is written very practical :)

**Comment by irodriguezclaveria** [comment]
Many thanks @ulie50, I'm going to take a look the link that you send me, and I think that is very good point to include the new category  market as "not disaster".

**Comment by irodriguezclaveria** [comment]
Hi @ulie50,

After to read the link that you send me, just I want to comment you, that  this steps that explain the article are already done with a train data, (tokenizer, clean stop words, TfidfTransformer …), thus I think that I'm going to continue checking and testing, and later I will try with another methods different to Naive Bayer to check that all is ok and can compare the accuracy.

**Comment by ulie50** [comment]
Hi Irene @irodriguezclaveria , yes, you did a very good job :) I just wanted to show where and for what data should be splited on two sets (again I havent found it in your code, may be I miss something). But anyhow for estimating we need to label the data.
